gridcomps/History/Sampler/MAPL_TrajectoryMod_smod.F90

```diff
@@ -915,28 +985,89 @@
+         chunk_field = ESMF_FieldCreate(this%LS_chunk,typekind=ESMF_TYPEKIND_R4,gridToFieldMap=[1],_RC)
+         call ESMF_FieldGet( src_field, localDE=0, farrayPtr=pt1, _RC )
+         call ESMF_FieldGet( chunk_field, localDE=0, farrayPtr=pt2, _RC )
+         pt1=0.0
+         pt2=0.0
+         call ESMF_FieldRedistStore(src_field,chunk_field,RH,_RC)
          call ESMF_FieldDestroy(src_field,noGarbage=.true.,_RC)
-         call ESMF_FieldDestroy(dst_field,noGarbage=.true.,_RC)
+         call ESMF_FieldDestroy(chunk_field,noGarbage=.true.,_RC)
 
          ! redist and put_var
          lm = this%vdata%lm
          acc_field_2d_rt = ESMF_FieldCreate (this%LS_rt, name='field_2d_rt', typekind=ESMF_TYPEKIND_R4, _RC)
          acc_field_3d_rt = ESMF_FieldCreate (this%LS_rt, name='field_3d_rt', typekind=ESMF_TYPEKIND_R4, &
               gridToFieldMap=[1],ungriddedLBound=[1],ungriddedUBound=[lm],_RC)
 
+         acc_field_2d_chunk = ESMF_FieldCreate (this%LS_chunk, name='field_2d_chunk', typekind=ESMF_TYPEKIND_R4, _RC)
+         acc_field_3d_chunk = ESMF_FieldCreate (this%LS_chunk, name='field_3d_chunk', typekind=ESMF_TYPEKIND_R4, &
+              gridToFieldMap=[1],ungriddedLBound=[1],ungriddedUBound=[lm],_RC)
+
+         !
+         !   caution about zero-sized array for MPI
+         !
+         nx_sum = this%nobs_epoch_sum
+         call ESMF_VMGetCurrent(vm,_RC)
+         call ESMF_VMGet(vm, mpiCommunicator=mpic, petCount=petCount, localPet=mypet, _RC)
+
+         iroot = 0
+         ip = mypet
+         N = nx_sum
+         M = petCount
+         nsend = int(ip+1, INT64) * int(N, INT64) / int(M, INT64) - &
+                 int(ip  , INT64) * int(N, INT64) / int(M, INT64)
+         allocate ( recvcount (petCount) )
+         allocate ( displs    (petCount) )
+         do ip=0, M-1
+            recvcount(ip+1) =  int(ip+1, INT64) * int(N, INT64) / int(M, INT64) - &
+                               int(ip  , INT64) * int(N, INT64) / int(M, INT64)
+         end do
+         displs(1)=0
+         do i = 2, petCount
+            displs(i) = displs(i-1) + recvcount(i-1)
+         end do
+
+         nsend_v = nsend * lm      ! vertical
+         allocate (recvcount_v, source = recvcount * lm )
+         allocate (displs_v, source = displs * lm )
+
+         if (mapl_am_i_root()) then
+            allocate ( p_acc_rt_2d(nx_sum) )
+         else
+            allocate ( p_acc_rt_2d(1) )
+         end if
+         !
+         ! p_dst (lm, nx)
+         if (mapl_am_i_root()) then
+            allocate ( p_acc_rt_3d(nx_sum,lm) )
+            allocate ( p_dst_rt(lm, nx_sum) )
+         else
+            allocate ( p_acc_rt_3d(1,lm) )
+            allocate ( p_dst_rt(lm, 1) )
+         end if
+
+#define lev_b_lev  1
+#if defined(lev_b_lev)
+         if (mapl_am_i_root()) write(6,*) 'lev b lev: gatherV ls_chunk to ls_root'
+#else
+         if (mapl_am_i_root()) write(6,*) '3d: gatherV ls_chunk to ls_root'
+#endif
+
```

Comment:
Why do we need a compile time switch here?  This seems wrong to me.

Reply:
I want to print out a single line into output to show which type of gatherV is used: level by level or 3D chunk. 
The main switch at line 1130 is located in the embedded do loop and putting print statement will generate more prints.  It is for debug purpose for real run, we donot need it (so I will delete it.)

Reply:
If you want to keep the diagnostic, this could be converted to a runtime conditional rather than a CPP conditional.